### PR TITLE
Updated API endpoints.

### PR DESCRIPTION
**Pull request by omkarjc27**
- Changed the API endpoints
- Changed the docs according to those endpoints.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,8 +70,8 @@ user_list is the list of local scores
 
 **Request :**
 ```
-curl base_url/user/create/ \
-    -X POST \
+curl base_url/users/ \
```
Comment (roymoran) [inline README.md:73]: @omkarjc27 we're going to start keeping tack of the api endpoints and their details in the docs directory. Take a look at my PR and introduce your changes for the interface there.

File: community.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 import json
```
Comment (roymoran) [inline community.py:6]: let's move all source files to a src directory

File: community.py [hunk 3]
```diff
@@ -37,13 +37,11 @@ def post(self):
 		return hsh_to_return
 
 class AddToCommunity(Resource):
-	def post(self):
+	def post(self,user_id):
 		# Inputs
-		API_data = request.get_json()
 		API_KEY = request.headers['THE_API_KEY']
```
Comment (roymoran) [inline community.py:42]: Are we validating the api key right now? If all endpoints will require an API key is there middlewear we can include that validates the key before reaching the logic of the endpoint? If many endpoints require the api key this seems like we'll be writing duplicate logic for each endpoint.

Comment (omkarjc27) [reply community.py:42]: I agree we should do that, 
Below are the code changes that we need to make.
- Interface with the db using an ORM (as discussed on slack).
- Create a global method to validate API-KEY.

File: README.md (hunk 1 above)
Comment (omkarjc27) [reply README.md:73]: I suggest we delete the README.md file in the root directory and keep the all documentation in docs/src/index.html.md . So there will be no need to update 2 docs when changes are made to the API.

File: README.md [hunk 4]
```diff
@@ -84,8 +84,8 @@ curl base_url/user/create/ \
 
 ### Create a Community
 ```
-curl base_url/community/create/ \
-    -X POST \
+curl base_url/community/ \
```
Comment (aaron2212) [inline README.md:87]: This should be `/communities/`

File: README.md [hunk 5]
```diff
@@ -84,8 +84,8 @@ curl base_url/user/create/ \
...
+    -X PUT \
```
Comment (aaron2212) [inline README.md:88]: If this is meant to be the endpoint for creating a user, it should be a `POST` request

File: README.md [hunk 6]
```diff
@@ -101,11 +101,9 @@ curl base_url/community/create/ \
 
 **Request :**
 ```
-curl base_url/community/add_user/ \
+curl base_url/community/<sample_user_name> \
```
Comment (aaron2212) [inline README.md:104]: Is this the endpoint for adding a user to a community? If that is the case, what do you guys @roy @omkarjc27 think about the endpoint being something like `/communities/:community_id/:user_id`?

File: README.md [hunk 7]
```diff
@@ -146,11 +144,9 @@ curl base_url/community/ \
 
 **Request :**
 ```
-curl base_url/user/award/ \
+curl base_url/award/<sample_user_name> \
```
Comment (aaron2212) [inline README.md:147]: `award` should be plural -> `awards`.

Is this the endpoint for getting awards for a specific user in a specific community? If so, then my comments is the as above @roymoran @omkarjc27

File: README.md [hunk 8]
```diff
@@ -165,7 +161,7 @@ curl base_url/user/award/ \
 
 **Request :**
 ```
-curl base_url/user/<username> -X GET
+curl base_url/users/<sample_user_name> -X GET
```
Comment (aaron2212) [inline README.md:164]: Question @roymoran and @omkarjc27 should we use `id` or `username` as the identifier? Does it matter?

File: README.md (hunk 6 above)
Comment (omkarjc27) [reply README.md:104]: That would surely make the method more readable.

But community name in the URL is not required as we will be able to identify the community based on the API KEY of that community.

File: README.md (hunk 5 above)
Comment (omkarjc27) [reply README.md:88]: Sure ✌️. Ill make that change.

File: src/community.py [hunk 9]
```diff
@@ -37,13 +37,11 @@ def post(self):
 		return hsh_to_return
 
 class AddToCommunity(Resource):
-	def post(self):
+	def post(self,user_id):
```
Comment (aaron2212) [inline src/community.py:40]: When adding a user to a community, the ID of the user should be sent in the request body, so the `user_id` parameter can be removed. I think the endpoint for adding a user to a community would look something like `/communities/:community_id/users/`

File: README.md (hunk 8 above)
Comment (omkarjc27) [reply README.md:164]: I guess username will be more easy for users to remember. I suggest **we should not** have an auto generated id for users. We should only generate the API key for communities.

File: src/main.py [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+from flask import Flask,request,render_template
+from flask_restful import Resource,Api
+from user import *
+from community import *
+
+app = Flask(__name__)
+api = Api(app)
+
+api.add_resource(ShowComm,'/community/') # GET request to get community info.
+api.add_resource(CreateComm,'/community/') # PUT request to create a community. Method for internal use only.
```
Comment (aaron2212) [inline src/main.py:11]: This should be a `POST` request, and `community` should be plural -> `communities`

File: src/main.py [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+api.add_resource(CreateUser,'/user/') # PUT request to create a user. Method for internal use only.
```
Comment (aaron2212) [inline src/main.py:12]: Same as here. `user` should be plural -> `users`

File: src/main.py [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+api.add_resource(ShowUser,'/user/<string:username>') # GET request to get user info.
+api.add_resource(AddToCommunity,'/add/<string:username>') # POST request to add a user to your community.
```
Comment (aaron2212) [inline src/main.py:14]: I think this could be like `/communities/<int:community_id>/users`

File: src/user.py [hunk 13]
```diff
@@ -6,7 +6,7 @@
 import json
 
 class CreateUser(Resource):
-	def post(self):
+	def put(self):
```
Comment (aaron2212) [inline src/user.py:9]: Creating a resource (user) is a `POST` request

File: README.md (hunk 5 above)
Comment (omkarjc27) [reply README.md:88]: The `create_user` and `create_community` methods are made to be accessed only by us (they wont be accessible to the public), so do we really need to follow conventions for these 2 methods ?

File: src/user.py (hunk 13 above)
Comment (omkarjc27) [reply src/user.py:9]: @aaron2212 My comments are as above.

File: src/community.py (hunk 9 above)
Comment (omkarjc27) [reply src/community.py:40]: @aaron2212  Is it safe to include community's API-KEY in the URL ?

File: src/main.py [hunk 14]
```diff
@@ -0,0 +1,18 @@
+import psycopg2
+from flask import Flask,request,render_template
+from flask_restful import Resource,Api
+from user import *
+from community import *
+
+app = Flask(__name__)
+api = Api(app)
+
+api.add_resource(ShowComm,'/community/') # GET request to get community info.
```
Comment (aaron2212) [inline src/main.py:10]: I'm not sure if Flask does this by default, but perhaps we should prefix these endpoints with `api/`

File: README.md (hunk 5 above)
Comment (aaron2212) [reply README.md:88]: It's always important to stick to conventions and be consistent through the application, so we should stick to the same conventions

File: src/main.py (hunk 14 above)
Comment (omkarjc27) [reply src/main.py:10]: I think we should prefix them by the API version `/v1/communities/` and then add api somewhere in the base url like `api.github.com`

File: README.md (hunk 1 above)
Comment (roymoran) [reply README.md:73]: We can keep readme.md but remove duplicate info. A couple thing we can update it with 
- Setting up and running dev environment
- Link to docs page

File: community.py (hunk 3 above)
Comment (roymoran) [reply community.py:42]: Alright that sounds good - it's okay if we address this in a follow up pr since this seems like a feature in itself.

**Pull request merged**